src/jit/compiler.h

```diff
@@ -3035,16 +3055,16 @@ class Compiler
     BOOL lvaIsOriginalThisReadOnly();           // return TRUE if there is no place in the code
                                                 // that writes to arg0
 
-    // Struct parameters that are passed by reference are marked as both lvIsParam and lvIsTemp
-    // (this is an overload of lvIsTemp because there are no temp parameters).
     // For x64 this is 3, 5, 6, 7, >8 byte structs that are passed by reference.
     // For ARM64, this is structs larger than 16 bytes that are passed by reference.
     bool lvaIsImplicitByRefLocal(unsigned varNum)
     {
 #if defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_)
-        LclVarDsc* varDsc = &(lvaTable[varNum]);
-        if (varDsc->lvIsParam && varDsc->lvIsTemp)
+        LclVarDsc* varDsc = lvaTable + varNum;
```

Comment:
What is a "recomended" way to access `LclVarDsc`? I think I prefer `LclVarDsc* lvaGetDesc(unsigned lclNum)`.

Reply:
I'd say it's preferable to use `lvaGetDesc`, at least because it contains an assert that checks for invalid variable numbers. But I'm probably biased since I added that function :)

If for some reason you can't/won't use the function then I'd stick to the `lvaTable[varNum]`. Somehow I don't fancy the unnecessary use of pointer arithmetic...